mlir/lib/Dialect/MIOpen/MIOpenDialect.cpp

```diff
@@ -442,6 +442,21 @@ LogicalResult InsertSliceOp::verify() {
 //===-----------------------------------------------------===//
 // TransformingForOp
 //===-----------------------------------------------------===//
+
+  /// Parse a list of assignments of the form
+  ///   (%x1 = %y1 : type1, %x2 = %y2 : type2, ...)
+static ParseResult
+  parseAssignmentListWithTypes(SmallVectorImpl<OpAsmParser::Argument> &lhs,
```

Comment:
This commit ad2af476ffcf7b09a2f2ba4b5648d851e7c21c75 sounds a bit suspicious. Why do we have to bring over a deleted method from upstream? Any other alternatives from `parser`?

Reply:
TransformingForOp::parse() needs this function.

Reply:
And the issue #596 will revisit this patch and see if we can do it without affecting the upstream.

Reply:
But there must be a reason upstream get rid of this. Are there no substitute left? And how does other places handle the mismatch when it also needs to parse something?

Reply:
I've found the commit and I can take the fix unless someone desperately wants a learning experience